sdk/src/azure/core/az_json_reader.c

```diff
@@ -164,7 +164,7 @@ AZ_NODISCARD static az_result _az_json_reader_get_next_buffer(
 
 AZ_NODISCARD static az_span _az_json_reader_skip_whitespace(az_json_reader* ref_json_reader)
 {
-  az_span json = { 0 };
+  az_span json;
```

Comment:
do we have to remove this? What I like about initializations, is that in case there's a bug, the behavior is consistent, so it shows up every time, rather than randomly.

Reply:
I fixed all instances of uninitialized values that the linter picked up.
Unlike the usual culprit which is common, this scenario is different.

A lot of times we have this scenario (`out_` parameters, for example), and I totally agree we should initialize the local here (the linter catches this).
**Bad:**
```C
int x;
some_method(..., &x);
```
**Good:**
```C
int x  = 0;
some_method(..., &x);
```

However, the linter also picks up redundant assignment, which is the precise case here and why I removed the initialization. In this case, we shouldn't be unnecessarily assigning a value that we aren't reading (`clang-analyzer-deadcode.DeadStore`).
**Bad:**
```C
int x = 0;
x = 5;
```
**Good:**
```C
int x;
x = 5;
```

Reply:
This is interesting.  I know there are some different views on the team with regards to initializing a variable at definition before passing it to a method that would initialize it.  (Non primitives like structs are different... they ought to use an init function and not have all member variables initialized individually before passing to an init function.)

@danewalton has a perspective on the advantage of gaining compiler warnings by not initializing at definition.

Another exception I've seen to this is the non-initialization of a return code or result at definition.  The advantage here is to avoid a developer error of not changing the variable to the correct value before returning it.  However, I can see this as a logic error that should be caught by a unit test.

Thoughts?